pkg/shell/hosts_dialog.jsx

```diff
@@ -475,7 +478,24 @@ class HostKey extends React.Component {
                 <ClipboardCopy isReadOnly hoverTip={_("Copy")} clickTip={_("Copied")} className="hostkey-fingerprint pf-v5-u-font-family-monospace">{fp}</ClipboardCopy>
                 <p className="hostkey-type">({key_type})</p>
                 <p>{cockpit.format(_("To verify a fingerprint, run the following on $0 while physically sitting at the machine or through a trusted network:"), this.props.host)}</p>
-                <ClipboardCopy isReadOnly hoverTip={_("Copy")} clickTip={_("Copied")} className="hostkey-verify-help-cmds pf-v5-u-font-family-monospace">ssh-keyscan -t {key_type} localhost | ssh-keygen -lf -</ClipboardCopy>
+                <ClipboardCopy isReadOnly hoverTip={_("Copy")} clickTip={_("Copied")} className="hostkey-verify-help-cmds pf-v5-u-font-family-monospace">{`ssh-keyscan -t ${key_type} localhost | ssh-keygen -lf -`}</ClipboardCopy>
+                <p>{_("Optionally, paste the result below to compare fingerprint")}</p>
+                <Form>
+                    <FormGroup>
+                        <TextArea id="command-ssh-keyscan-result-paste" spellCheck="false" placeholder="# localhost:22 SSH-..."
+                                    validated={this.state.fpValidated}
+                                    onChange={(_ev, remote_fp) => { this.setState({ fpValidated: remote_fp ? remote_fp.match(fp) ? "success" : "error" : undefined }) } } />
+                        {this.state.fpValidated && (
+                            <FormHelperText>
+                                <HelperText>
+                                    <HelperTextItem variant={this.state.fpValidated}>
+                                        {this.state.fpValidated == 'success' ? 'Fingerprint matched' : 'Fingerprint not present'}
```

Comment:
Shouldn't this say "Fingerprint does not match" instead of "Fingerprint not present"?

Reply:
Well, user can paste any string, it is possible for no fingerprint to be present.

Either
1. add another check to match any fingerprint and then tell the fingerprint does not match.
2. or just replace "Fingerprint not present" with "Required fingerprint not present"

I'd vote for adding another check, may update in the next commit.

Reply:
Ah, I see, hmm. Yeah, if we can detect whether or not there is a fingerprint in the textarea, that would be awesome. If not, I think I prefer the error message to say "Fingerprints do not match". @garrett , what do you think?